code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm

```diff
@@ -65,37 +65,37 @@
 	fallback_specific_heat = 0.018
 	scannable = TRUE
 
-/datum/reagent/carbon/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
+/decl/reagent/carbon/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagents/holder)
 	var/datum/reagents/ingested = M.get_ingested_reagents()
-	if(ingested && ingested.reagent_list.len > 1) // Need to have at least 2 reagents - cabon and something to remove
-		var/effect = 1 / (ingested.reagent_list.len - 1)
-		for(var/datum/reagent/R in ingested.reagent_list)
-			if(R == src)
+	if(LAZYLEN(ingested?.reagent_volumes) > 1) // Need to have at least 2 reagents - cabon and something to remove
```

Comment:
`LAZYLEN` should be used to provide the length of a list OR null if the list is null. It shouldn't be used to test if a datum exists. You're doing this in a bunch of places. Should be replaced with:

```suggestion
	if(ingested && LAZYLEN(ingested.reagent_volumes) > 1) // Need to have at least 2 reagents - cabon and something to remove
```

And similar in all other places that we're unsure if the datum exists.

Reply:
We don't do `x && istype(x)`, I fail to see how `x && LAZYLEN(x)` (or, related, `LAZYLEN(y?.x)`) qualifies.

Reply:
Discussed on Discord why, hope that made sense for you.